internal/services/recoveryservices/recovery_services_vault_resource.go

```diff
@@ -189,18 +188,12 @@ func resourceRecoveryServicesVault() *pluginsdk.Resource {
 func resourceRecoveryServicesVaultCreate(d *pluginsdk.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).RecoveryServices.VaultsClient
 	cfgsClient := meta.(*clients.Client).RecoveryServices.VaultsConfigsClient
-	storageCfgsClient := meta.(*clients.Client).RecoveryServices.StorageConfigsClient
 	settingsClient := meta.(*clients.Client).RecoveryServices.VaultsSettingsClient
 	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
 	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
 	id := vaults.NewVaultID(subscriptionId, d.Get("resource_group_name").(string), d.Get("name").(string))
-	storageId := backupresourcestorageconfigsnoncrr.VaultId{
-		SubscriptionId:    id.SubscriptionId,
-		ResourceGroupName: id.ResourceGroupName,
-		VaultName:         id.VaultName,
-	}
 	cfgId := backupresourcevaultconfigs.VaultId{
```

Comment:
I realise this code was already present in the Provider, but can we fix this to use the `New` method?

```suggestion
	cfgId := backupresourcevaultconfigs.NewVaultID(
```